snap/hooks/install

```diff
@@ -7,43 +7,53 @@ source "${OPS_ROOT}"/helpers/snap-logger.sh "hook-install"
 source "${OPS_ROOT}"/helpers/io.sh
 source "${OPS_ROOT}"/helpers/set-conf.sh
 
-
 function create_file_structure () {
-    add_folder "${SNAP_COMMON}/backups" 774
-    ls -la "${SNAP_COMMON}"
-    ls -la "${SNAP_COMMON}/backups"
-    dir_copy_if_not_exists "config" "${SNAP_COMMON}/backups" 774
-
-    # -------------------------------
-
-    # Opensearch install folders and set config
-    dir_copy_if_not_exists "bin" "${SNAP_DATA}" 770
-    dir_copy_if_not_exists "jdk" "${SNAP_DATA}" 770
-    dir_copy_if_not_exists "lib" "${SNAP_DATA}" 550
-    dir_copy_if_not_exists "modules" "${SNAP_DATA}" 550
-    dir_copy_if_not_exists "plugins" "${SNAP_DATA}" 770
-
-    dir_copy_if_not_exists "config" "${SNAP_DATA}" 770
-    add_folder "${SNAP_DATA}/config/certificates" 770
-    add_file "${SNAP_DATA}/config/unicast_hosts.txt" 770
-
-    add_folder "${SNAP_DATA}/extensions" 770
-    add_file "${SNAP_DATA}/extensions/extensions.yml" 770
-
-    add_folder "${SNAP_COMMON}/data" 770
-    add_folder "${SNAP_COMMON}/logs" 774
-    add_folder "${SNAP_COMMON}/tmp" 770
+    mkdir -p "${OPENSEARCH_PATH_CONF}"
+    mkdir -p "${OPENSEARCH_HOME}"
+    mkdir -p "${OPENSEARCH_VARLIB}"
+    mkdir -p "${OPENSEARCH_VARLOG}"
+    mkdir -p "${OPENSEARCH_TMPDIR}"
+    mkdir -p "${OPENSEARCH_PATH_CERTS}"
+    mkdir -p "${OPENSEARCH_PATH_CONF}/extensions"
+    mkdir -p "${SNAP_COMMON}/backups"
+
+    chmod -R 774 "${SNAP_COMMON}/backups"
+
+    cp -n -r "${SNAP}"/config/* "${OPENSEARCH_PATH_CONF}"
 }
 
 
 function set_base_config_props () {
     # Change conf to set log and data paths for both opensearch.yaml and jvm.options
-    set_yaml_prop "${SNAP_DATA}/config/opensearch.yml" "path.data" "${SNAP_COMMON}/data"
-    set_yaml_prop "${SNAP_DATA}/config/opensearch.yml" "path.logs" "${SNAP_COMMON}/logs"
-
-    replace_in_file "${SNAP_DATA}/config/jvm.options" "=logs/" "=${SNAP_COMMON}/logs/"
+    #     More info, check: https://github.com/opensearch-project/OpenSearch/blob/  \
+    #     418ab51718cb68b7a57d4cf524b1dec9ed02b224/server/src/main/java/org/opensearch/env/Environment.java#L68
+    set_yaml_prop "${OPENSEARCH_PATH_CONF}/opensearch.yml" "path.data" "${OPENSEARCH_VARLIB}"
+    set_yaml_prop "${OPENSEARCH_PATH_CONF}/opensearch.yml" "node.pidfile" "${OPENSEARCH_HOME}"/pidfile
```

Comment:
Why is this needed?

Reply:
So, the ```path.data``` and the ```path.logs``` point to folders within ```SNAP_COMMON``` whereas ```OPENSEARCH_HOME``` points to ```SNAP_DATA```, so these need to be overriden.

The pidfile, on the other hand, is a marker that opensearch is running on that node. Setting the pidfile allows to quickly check if we can start a new OpenSearch process or not in the same host, for the same snap. I'd rather have it and just ignore if we do not use for anything (e.g. in the systemd file).
If we leave ```node.pidfile``` unset, then, [pidfile is not created](https://github.com/opensearch-project/OpenSearch/blob/ca74aac1482718a864676e27707bb77b7c168e6f/server/src/main/java/org/opensearch/bootstrap/Bootstrap.java#L377-L383)

Reply:
Okay, this is fine. We'll see how it behaves in the charm, where we test on the status of the service with:
`systemctl status snap.opensearch.daemon`